include/wobbly/geometry.h

```diff
@@ -0,0 +1,352 @@
+                    return *this;
+                }
+
+                /* cppcheck thinks this is a constructor, so we need to suppress
+                 * a constructor warning here.
+                 */
+                // cppcheck-suppress uninitMemberVar
+                operator PointView <NTC> ()
+                {
+                    /* The stored offset is an array offset, but the constructor
+                     * requires a point offset. Divide by two */
+                    return PointView <NTC> (array, offset / 2);
+                }
+
+                template <size_t Offset>
+                NumericType const & get () const
+                {
+                    static_assert (Offset < 2, "Offset must be < 2");
+                    return array[offset + Offset];
+                }
+
+                template <size_t Offset>
+                typename std::enable_if <!(IC::value) && (Offset + 1), void>::type
+                set (NumericType value)
+                {
+                    static_assert (Offset < 2, "Offset must be < 2");
+                    array[offset + Offset] = value;
+                }
+
+            private:
+
+                NumericType *array;
+                size_t      offset;
+        };
+
+        namespace dimension
+        {
+            /* Overloads for wobbly::PointView */
+            template <typename U>
+            struct Dimension <PointView <U> >
+            {
+                typedef double data_type;
+                static const size_t dimensions = 2;
+            };
+
+            template <typename T, size_t D>
+            struct DimensionAccess <PointView <T>, D>
+            {
+                static inline T get (PointView <T> const &p)
+                {
+                    return p.template get <D> ();
+                }
+
+                static inline void
+                set (PointView <T> &p, T const &value)
+                {
+                    p.template set <D> (value);
+                }
+            };
+        }
+
+        /**
+         * PointModel<T>:
+         *
+         * A detached 2D point or vector in space for a given data
+         * type T. This is a structure of two values.
+         *
+         * PointModel implements the Dimension trait, meaning that it
+         * can be used with functions in the wobbly::geometry::dimension
+         * namespace.
+         */
+        template <typename T>
+        struct PointModel
+        {
+            PointModel (T x, T y) noexcept :
+                x (x),
+                y (y)
+            {
+            }
+
+            PointModel () noexcept :
+                x (0),
+                y (0)
+            {
+            }
+
+            PointModel (PointModel const &p) noexcept :
+                x (p.x),
+                y (p.y)
+            {
+            }
+
+            void swap (PointModel &a, PointModel &b) noexcept
+            {
+                using std::swap;
+
```

Comment:
Again, I'd prefer to use the namespace here. The use of the newline also differs from the prevuous swap where there was none.

Reply:
Sure, make sense